src/main/java/com/google/api/codegen/transformer/MethodTransformerContext.java

```diff
@@ -79,11 +80,31 @@ public InterfaceConfig getInterfaceConfig() {
   }
 
   public Collection<CollectionConfig> getCollectionConfigs() {
-    return getInterfaceConfig().getCollectionConfigs();
+    return getApiConfig().getCollectionConfigs().values();
   }
 
   public CollectionConfig getCollectionConfig(String entityName) {
-    return getInterfaceConfig().getCollectionConfig(entityName);
+    return getApiConfig().getCollectionConfig(entityName);
+  }
+
+  public boolean entityNameIsCollectionConfig(String entityName) {
+    return getApiConfig().getCollectionConfig(entityName) != null;
+  }
+
+  public boolean entityNameIsCollectionOneofConfig(String entityName) {
+    return getApiConfig().getCollectionOneofConfigs().get(entityName) != null;
+  }
+
+  public CollectionConfig getFirstCollectionConfig(String entityName) {
```

Comment:
That actually does't address the "why" though - what makes it so that the "first" collection config found is the correct one for the use case?